tests/index.ts

```diff
@@ -119,6 +120,69 @@ export const setupDriver = (
   return driver;
 };
 
+export const setupTrunkToolDriver = (
+  dirname: string,
+  { setupGit = true, setupTrunk = true, trunkVersion = undefined }: SetupSettings,
+  toolName?: string,
+  version?: string
+): TrunkToolDriver => {
+  const driver = new TrunkToolDriver(
+    dirname,
+    { setupGit, setupTrunk, trunkVersion },
+    toolName,
+    version
+  );
+
+  beforeAll(async () => {
+    await driver.setUp();
+  });
+
+  afterAll(() => {
+    driver.tearDown();
+  });
+
+  afterEach(() => {
+    registerVersion(driver.enabledVersion);
+  });
+  return driver;
+};
+
+// Tool tests:
+// should take in: tool name
+// dir name
+// input/output
+
+interface ToolTestConfig {
+  command: string[];
+  expectedOut: string;
+  expectedErr: string;
+  expectedExitCode: number;
+}
+
+export const toolTest = ({
+  toolName,
+  toolVersion,
+  testConfigs,
+  dirName = path.dirname(caller()),
+}: {
+  toolName: string;
+  toolVersion: string;
+  dirName?: string;
+  testConfigs: ToolTestConfig[];
+}) => {
+  describe(toolName, () => {
+    const driver = setupTrunkToolDriver(dirName, {}, toolName, toolVersion);
+    testConfigs.forEach(({ command, expectedOut, expectedErr, expectedExitCode }) => {
+      it(`should run "${command.join(" ")}" and exit with code ${expectedExitCode}`, async () => {
+        const { stdout, stderr, exitCode } = await driver.runTool(command);
+        expect(stdout).toContain(expectedOut);
+        expect(stderr).toContain(expectedErr);
+        expect(exitCode).toEqual(expectedExitCode);
```

Comment:
I would recommend a custom matcher here, since it can give more information at once in the object. Rn, if this fails, expect only tells you the first failure. There's objectContains as well, which might also be a fit, although it doesn't support substrings out of the box. Can be tabled for later if you want